components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/CertificateValidationUtil.java

```diff
@@ -1223,4 +1237,184 @@ private static boolean isSelfSignedCert(X509Certificate cert) {
         }
     }
 
+    private static void addDefaultValidatorConfigToConfigStore(OMElement validatorsElement, String tenantDomain)
+            throws CertificateValidationException {
+
+        boolean isAdded = addValidatorResourceType(tenantDomain);
+        if (!isAdded) {
+            log.debug("Validator type is already added to the tenant config store. Skip adding validator configs");
+            return;
+        }
+
+        List<Validator> defaultValidatorConfig = getDefaultValidatorConfig(validatorsElement);
+        for (Validator validator : defaultValidatorConfig) {
+            addValidatorResource(validator);
+        }
+    }
+
+    private static boolean addValidatorResourceType(String tenantDomain) throws CertificateValidationException {
+
+        boolean isAdded = false;
```

Comment:
Since we need to add this Resource Type only for an one time Shall we add this resource type using the SQL script.

https://github.com/wso2/carbon-identity-framework/blob/13c51093d7d2d74a6afd2caeac459d13e496f1d3/features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/h2.sql#L973-L988